internal/oauth/oauth.go

```diff
@@ -0,0 +1,190 @@
+package oauth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/foundriesio/fioctl/internal/config"
+	"github.com/sirupsen/logrus"
+)
+
+const uri string = "https://app.foundries.io/oauth"
+
+// OAuth interface.
+type OAuth interface {
```

Comment:
Any reason you are doing the interface versus just returning ClientCredentials? ie - its exposing the same methods (since they are UpperCase). I guess it also has "Config" and "URL", but you could make those lowercase and get the same effect.

Reply:
My initial idea was to abstract on that in order to potentially handle other OAuth credentials flow, like the device registration, that have slightly different flows but could share a common interface.